cmd/nerdctl/pull.go

```diff
@@ -92,6 +100,46 @@ func pullAction(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	if isVerify, err := cmd.Flags().GetString("verify"); err == nil && isVerify == "cosign" {
+		cosignExecutable, err := exec.LookPath("cosign")
+		if err != nil {
+			logrus.WithError(err).Error("cosign executable not found in path $PATH")
+			logrus.Info("you might consider installing cosign from: https://docs.sigstore.dev/cosign/installation")
+			return err
+		}
+
+		cosignCmd := exec.Command(cosignExecutable, []string{"verify"}...)
+		cosignCmd.Env = os.Environ()
+
+		keyRef, err := cmd.Flags().GetString("cosign-key")
+		if err != nil {
+			return err
+		}
+
+		if keyRef != "" {
+			cosignCmd.Args = append(cosignCmd.Args, "--key", keyRef)
+		} else {
+			cosignCmd.Env = append(cosignCmd.Env, "COSIGN_EXPERIMENTAL=true")
+		}
+
+		cosignCmd.Args = append(cosignCmd.Args, rawRef)
+
+		logrus.Debugf("running %s %v", cosignExecutable, cosignCmd.Args)
+
+		stdout, _ := cosignCmd.StdoutPipe()
+		if err := cosignCmd.Start(); err != nil {
+			return err
+		}
+
+		scanner := bufio.NewScanner(stdout)
+		for scanner.Scan() {
+			logrus.Info(scanner.Text())
```

Comment:
Perhaps `logrus.Info("cosign: " + scanner.Text())` to clarify that the output is from cosign, not really from nerdctl.

Reply:
it makes sense, thanks, done.